library/src/main/java/com/novoda/downloadmanager/lib/DownloadThread.java

```diff
@@ -341,6 +345,10 @@ private void executeDownload(State state) throws StopRequestException {
         state.resetBeforeExecute();
         setupDestinationFile(state);
 
+        if (originalDownloadInfo.shouldAllowTarUpdate(state.mimeType)) {
+            state.totalBytes = -1;
```

Comment:
my changes we super basic https://github.com/novoda/download-manager/pull/118 there will be no problems